pkg/component/etcd/configmap/configmap.go

```diff
@@ -200,54 +138,63 @@ func (c *component) syncConfigmap(ctx context.Context, cm *corev1.ConfigMap) err
 		return err
 	}
 	c.values.ConfigMapChecksum = utils.ComputeSHA256Hex(jsonString)
-
-	return nil
-
-}
-
-func (c *component) Destroy(ctx context.Context) error {
-	configMap := c.emptyConfigmap(c.values.ConfigMapName)
-
-	if err := c.deleteConfigmap(ctx, configMap); err != nil {
-		return err
-	}
 	return nil
 }
 
 func (c *component) deleteConfigmap(ctx context.Context, cm *corev1.ConfigMap) error {
 	return client.IgnoreNotFound(c.client.Delete(ctx, cm))
 }
 
-func (c *component) emptyConfigmap(name string) *corev1.ConfigMap {
+func (c *component) emptyConfigmap() *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
+			Name:      c.values.Name,
 			Namespace: c.namespace,
 		},
 	}
 }
 
-func getObjectMeta(val *Values) metav1.ObjectMeta {
-	labels := map[string]string{"name": "etcd", "instance": val.EtcdName}
-	for key, value := range val.Labels {
-		labels[key] = value
-	}
-	return metav1.ObjectMeta{
-		Name:      val.ConfigMapName,
-		Namespace: val.EtcdNameSpace,
-		Labels:    labels,
-	}
+type etcdConfig struct {
+	Name                    string         `yaml:"name"`
+	DataDir                 string         `yaml:"data-dir"`
+	Metrics                 string         `yaml:"metrics"`
+	SnapshotCount           int            `yaml:"snapshot-count"`
+	EnableV2                bool           `yaml:"enable-v2"`
+	QuotaBackendBytes       int64          `yaml:"quota-backend-bytes"`
+	InitialClusterToken     string         `yaml:"initial-cluster-token"`
+	InitialClusterState     string         `yaml:"initial-cluster-state"`
+	InitialCluster          string         `yaml:"initial-cluster"`
+	AutoCompactionMode      string         `yaml:"auto-compaction-mode"`
+	AutoCompactionRetention string         `yaml:"auto-compaction-retention"`
+	ListenPeerUrls          string         `yaml:"listen-peer-urls"`
+	ListenClientUrls        string         `yaml:"listen-client-urls"`
+	AdvertisePeerUrls       string         `yaml:"initial-advertise-peer-urls"`
+	AdvertiseClientUrls     string         `yaml:"advertise-client-urls"`
+	ClientSecurity          securityConfig `yaml:"client-transport-security,omitempty"`
+	PeerSecurity            securityConfig `yaml:"peer-transport-security,omitempty"`
 }
```

Comment:
Consider moving this type definition to the start of the file or at least above it's first reference

Reply:
Thanks for the review @aaronfern .

The idea behind defining the `etcdConfig` struct after the component struct and methods was to maintain a flow in the codebase, where the higher-level components and their associated methods are presented first, and then we delve into the specifics like the `etcdConfig` struct.

If you feel that a different ordering would improve readability or provide better consistency with the rest of the project, I'm open to making those changes. Let me know what you think.